src/ergw_config.erl

```diff
@@ -119,6 +125,21 @@ validate_node_name(Name)
 validate_node_name(Name) ->
     throw({error, {options, Name}}).
 
+validate_teid({ok, {Prefix, Len} = WrongConfig}, DefaultConfig)
+	when not is_integer(Len); Len < 0; Len > ?MAX_TRIES; not is_integer(Prefix); Prefix < 0; Prefix > 8 ->
+		?LOG(warning, "TEID length and/or prefix has wrong configuration: ~p", [WrongConfig]),
+		DefaultConfig;
+validate_teid({ok, {Prefix, Len}}, _) ->
+	{Prefix, Len};
+validate_teid({ok, {Len}}, DefaultConfig) when not is_integer(Len); Len < 0; Len > ?MAX_TRIES ->
+	?LOG(warning, "TEID length has wrong configuration: ~p", [Len]),
+	DefaultConfig;
+validate_teid({ok, {Len}}, _) ->
+	{(fun(X) when X =< 3 -> ?POW(2, X); (_) -> 1 end)(Len), Len};
+validate_teid(WrongConfig, DefaultConfig) ->
+	?LOG(warning, "TEID has wrong configuration: ~p", [WrongConfig]),
+	DefaultConfig.
+
```

Comment:
validations happens before application startup and not later.
Please make this part of the already existing validation runs.